### PR TITLE
baton-bitbucket-datacenter fixes

### PR DESCRIPTION
**Pull request by mchavez**
# Pull Request

To help us get this pull request reviewed and merged quickly, please be sure to include the following items:

* [ ] Tests (if applicable)
* [ ] Documentation (if applicable)
* [ ] Changelog entry
* [ ] A full explanation here in the PR description of the work done

## PR Type
What kind of change does this PR introduce?
- Skip global permission when using token
- Remove group trait when provisioning repo permissions
- Remove group trait when provisioning project permissions

* [X] Bugfix
* [ ] Feature
* [ ] Code style update (formatting, local variables)
* [X] Refactoring (no functional changes, no api changes)
* [ ] Build related changes
* [ ] CI related changes
* [ ] Documentation content changes
* [ ] Tests
* [ ] Other

## Backward Compatibility

Is this change backward compatible with the most recently released version? Does it introduce changes which might change the user experience in any way? Does it alter the Connector in any way?

* [ ] Yes (backward compatible)
* [ ] No (breaking changes)

## What's new?
- Skip global permission when using token
- Remove group trait when provisioning repo permissions
- Remove group trait when provisioning project permissions

### CODE REVIEW

File: pkg/connector/group.go [hunk 1]
```diff
@@ -101,7 +105,11 @@ func (g *groupBuilder) Entitlements(ctx context.Context, resource *v2.Resource,
 		Page:    pageToken,
 	})
 	if err != nil {
-		return nil, "", nil, err
+		if !strings.Contains(err.Error(), Unauthenticated) {
```
Comment (ggreer) [inline pkg/connector/group.go:108]: You can check the status code on the error with something like:
```go
		if status.Code(err) != codes.Unauthenticated {
			...
		}
```

You'll need to import these:
```
	"google.golang.org/grpc/status"
	"google.golang.org/grpc/codes"
```

Comment (mchavez) [reply pkg/connector/group.go:108]: Thanks for sharing the code. It looks nice and neat. However, in this case, status.Code(err) returns "Unknown" and codes.Unauthenticated = 16

Comment (mchavez) [reply pkg/connector/group.go]: Fixed

File: pkg/client/client.go [hunk 2]
```diff
@@ -605,41 +655,47 @@ func (d *DataCenterClient) GetGlobalGroupPermissions(ctx context.Context, startP
...
 
-func (d *DataCenterClient) ListGlobalGroupPermissions(ctx context.Context, opts PageOptions) ([]GroupsPermissions, string, error) {
+func (d *DataCenterClient) ListGlobalGroupPermissions(ctx context.Context, opts PageOptions) ([]GroupsPermissions, string, BitbucketError) {
 	var nextPageToken string = ""
 	groupsPermissions, page, err := d.GetGlobalGroupPermissions(ctx,
 		strconv.Itoa(opts.Page),
 		strconv.Itoa(opts.PerPage),
 	)
-	if err != nil {
+	if err.Error != nil {
```
Comment (ggreer) [inline pkg/client/client.go:688]: If there's no error, won't err be nil and cause this to crash?

Comment (mchavez) [reply pkg/client/client.go]: Fixed

**Pull request merged**